### DISCUSSION TITLE
Strange network traffic from clients to server after booting up class - even without login.

### DISCUSSION THREAD

**Discussion by abixadamj** [ltsp/ltsp#315] [Q&A]
I have strange situation. When I start class (20 PC) with Epoptes, PC's are booting up. Gets image from server, load it and display MDM (It is LTSP 20 based on Linux Mint 19.3) and waiting for login. No person goes to PC's, no one is trying to login. So, the network traffic should be very tittle. But it does not. 

![obraz](https://user-images.githubusercontent.com/5088643/94285609-2f0df100-ff54-11ea-8097-1efc2f546faf.png)

Client are sending a lot of traffic to server for 2-3 minutes. and after that, everything goes to "normal" traffic. Maybe someone knows, where shoud I look for anything, which can cause such situation?
Adam

**Comment by alkisg** [comment]
This might be an update service, like `apt update` or `snap update`.
LTSP tries to block most of them, but if you have one that's not in the list, you can put it in MASK_SYSTEM_SERVICES (see `man ltsp.conf`).

Open a root terminal with epoptes, and run:

```shell
root@pc01:~# df -h
Filesystem               Size  Used Avail Use% Mounted on
udev                     1.5G     0  1.5G   0% /dev
tmpfs                    300M  2.6M  297M   1% /run
tmpfs                    1.5G  6.1M  1.5G   1% /run/initramfs/ltsp
/dev/loop0p1             9.8G  7.8G  1.5G  84% /run/initramfs/ltsp/0/looproot
/run/initramfs/ltsp      1.5G  6.1M  1.5G   1% /
...
```
The /run/initramfs/ltsp line there means that the tmpfs RAM usage is 6.1 MB, that only 6 MB were written to the disk while it was booted.
If you see something like 50 MB or 100 MB, then that's probably the issue.
You can then run something like the following, to see which files exactly were written:

```shell
root@pc01:~# du -sh /run/initramfs/ltsp/0/up/*
4.0K    /run/initramfs/ltsp/0/up/boot
120K    /run/initramfs/ltsp/0/up/etc
0       /run/initramfs/ltsp/0/up/media
4.0K    /run/initramfs/ltsp/0/up/root
4.0K    /run/initramfs/ltsp/0/up/tmp
244K    /run/initramfs/ltsp/0/up/usr
5.8M    /run/initramfs/ltsp/0/up/var
```

That last line there means that files inside "/var" were written; if I continue with `du -sh` inside the var directory, I will be able to pinpoint the exact files that take up most of the space, and from there, find the service responsible.

**Comment by abixadamj** [comment]
OK, @alkisg , I know what was that. My client installed  on the server Moodle instance, and mysql database was the traffic.

![Zrzut ekranu 2020-10-4 o 21 34 44](https://user-images.githubusercontent.com/5088643/95576654-1829cd00-0a31-11eb-8891-b2a3d58fb880.png)


So I exclude `mysql` directory and now is OK.

----

I will just ask you for:
`/run/initram/ltsp` - which is **18 GB** big. What is this?
![Zrzut ekranu 2020-10-5 o 09 38 30](https://user-images.githubusercontent.com/5088643/95576756-41e2f400-0a31-11eb-938a-f2c35ee1c838.png)

**Comment by alkisg** [comment]
/run/initramfs/ltsp/0/looproot is the mount point of your virtual disk, i.e. of the server /srv/ltsp/images/x86_64.img. That's why it's big.
The up/ and work/ directories are where the overlayfs stores the tmpfs writes after the client boots. That's why they're small.

**Comment by abixadamj** [comment]
@alkisg - Thank you for help. Everything with this issue is now OK.
